### DISCUSSION TITLE
Make implicit void redirect the context to the last explicit context

### DISCUSSION THREAD

**Discussion by HellGate94** [dotnet/csharplang#5984] [Language Ideas] [answered]
Given this class

```
class Foo {
    public void A() { }
    public void B() { }
}
Foo foo = new();
```

and the usage:
`foo.A().B();`,

will result in a `CS0023: Operator '.' cannot be applied to operand of type 'void'`
where the equivalent code would be `foo.A(); void.B();`

however if implicit void types (return type of `A()` would be the implicit void type) redirect its context to the previous explicit context (`foo` in this case), the previous example would compile just fine as `B()` would now point back to `foo` resulting in the equivalent code of `foo.A(); foo.B();`

Main advantages: This change would enable fluent syntax at a language level. Libraries old and new no longer need to explicitly support it

Disadvantages: Adds quite some language complexity and it can be confusing to know on what context you are working on without good IDE integration (something similar to bracket pair highlight)

**Comment by Joe4evr** [comment]
https://github.com/dotnet/csharplang/discussions/74 https://github.com/dotnet/csharplang/discussions/96

**Reply by HellGate94** [reply]
these seem to focus on moving method parameters out on their own lines where as mine is purely for chaining methods as if they would do
`return this;`

**Comment by jnm2** [comment, marked as answer]
Duplicate of: https://github.com/dotnet/csharplang/discussions/781

Solving the problem other ways: https://github.com/dotnet/csharplang/issues/311, https://github.com/dotnet/csharplang/discussions/577, https://github.com/dotnet/csharplang/discussions/1342, https://github.com/dotnet/csharplang/discussions/2642
